opencompass/summarizers/multi_model.py

```diff
@@ -0,0 +1,374 @@
+            for model_abbr in model_abbrs:
+                if dataset_abbr in parsed_results[model_abbr]:
+                    row.append('{:.02f}'.format(parsed_results[model_abbr][dataset_abbr][index]))
+                else:
+                    row.append('-')
+            table.append(row)
+
+        self.models_summary_group_metrics[table[0][-1]] = summary_group_metrics
+        return table
+
+    def merge(self, summarizer: 'MultiModelSummarizer'):
+        assert len(self.table) == len(summarizer.table)
+        for row_i, row in enumerate(summarizer.table):
+            base_row = self.table[row_i]
+            if base_row[:3] != row[:3]:
+                self.logger.warning(f'cannot merge tables with different headers: {base_row} vs {row}')
+            base_row.append(row[-1])
+        new_model_name = summarizer.table[0][-1]
+        assert new_model_name not in self.models_summary_group_metrics
+        self.models_summary_group_metrics[new_model_name] = summarizer.models_summary_group_metrics[new_model_name]
+
+    def summarize(self):
+        """
+        Format in self.table
+        [
+            ['dataset', 'version', 'metric', 'mode', 'model_name'],
+            ['--------- 考试 Exam ---------', '-', '-', '-', '-'],
+            ['ARC-c', '1e0de5', 'accuracy', 'gen', '79.32'],
+            ['ARC-e', '1e0de5', 'accuracy', 'gen', '85.36'],
+            ['--------- 语言 Language ---------', '-', '-', '-', '-'],
+            ['WiC', 'd06864', 'accuracy', 'gen', '55.64'],
+            ['chid-dev', '211ee7', 'accuracy', 'gen', '52.97'],
+            ['--------- 知识 Knowledge ---------', '-', '-', '-', '-'],
+            ['BoolQ', '883d50', 'accuracy', 'gen', '86.06'],
+            ['--------- 理解 Understanding ---------', '-', '-', '-', '-'],
+            ['C3', '8c358f', 'accuracy', 'gen', '88.33'],
+            ['race-middle', '9a54b6', 'accuracy', 'gen', '90.32'],
+            ['--------- 推理 Reasoning ---------', '-', '-', '-', '-'],
+            ['cmnli', '1abf97', 'accuracy', 'gen', '38.26'],
+            ['ocnli', 'c4cb6c', 'accuracy', 'gen', '32.92'],
+        ]
+        """
+
+        table = Table()
+        for i, col_name in enumerate(self.table[0]):
+            table.add_column(col_name, overflow='fold', max_width=20 if i >= META_COL_COUNT else None)
+
+        section_rows = []
+        all_rows = []
+        for row_i, row in enumerate(self.table):
+            if row_i == 0:
+                continue
+            if is_section_row(row):
+                table.add_section()
+                new_row = create_section_row(row_i, row, self.table)
+                section_rows.append(new_row)
+            else:
+                new_row = row
+                all_rows.append(new_row)
+
+            table.add_row(*highlight(new_row))
+
+        table.add_section()
+        average_row = average_rows('Naive Average', section_rows)
+        average_row = highlight(average_row)
+        table.add_row(*average_row)
```

Comment:
section rows might also not exists for singular dataset eval
```suggestion
        if section_rows:
            table.add_section()
            average_row = average_rows('Naive Average', section_rows)
            average_row = highlight(average_row)
            table.add_row(*average_row)
```